Documentation/Rendering/ClippingPrimitive.md

```diff
@@ -0,0 +1,114 @@
+
+> [!Note]
+> Having multiple [`ClippingPrimitives`](xref:Microsoft.MixedReality.Toolkit.Utilities.ClippingPrimitive) clip a [renderer](https://docs.unity3d.com/ScriptReference/Renderer.html) will increase pixel shader instructions and will impact performance. Please profile these changes within your project.
+
+*How to have two different [`ClippingPrimitives`](xref:Microsoft.MixedReality.Toolkit.Utilities.ClippingPrimitive) clip a render. For example a [`ClippingSphere`](xref:Microsoft.MixedReality.Toolkit.Utilities.ClippingSphere) and [`ClippingBox`](xref:Microsoft.MixedReality.Toolkit.Utilities.ClippingBox) at the same time:*
+
+```C#
+// Within MRTK/Core/StandardAssets/Shaders/MixedRealityStandard.shader (or another MRTK shader) change:
+
+#pragma multi_compile _ _CLIPPING_PLANE _CLIPPING_SPHERE _CLIPPING_BOX
+
+// to:
+
+#pragma multi_compile _ _CLIPPING_PLANE
+#pragma multi_compile _ _CLIPPING_SPHERE
+#pragma multi_compile _ _CLIPPING_BOX
+```
+> [!Note]
+> The above change will inccur additional shader compilation time.
+
+*How to have two of the same [`ClippingPrimitives`](xref:Microsoft.MixedReality.Toolkit.Utilities.ClippingPrimitive) clip a render. For example two [`ClippingBoxes`](xref:Microsoft.MixedReality.Toolkit.Utilities.ClippingBox) at the same time:*
+
+```C#
+// 1) Add the below MonoBehaviour to your project:
+
+[ExecuteInEditMode]
+public class SecondClippingBox : ClippingBox
+{
+    /// <inheritdoc />
+    protected override string Keyword
+    {
+        get { return "_CLIPPING_BOX2"; }
+    }
+
+    /// <inheritdoc />
+    protected override string ClippingSideProperty
+    {
+        get { return "_ClipBoxSide2"; }
+    }
+
+    /// <inheritdoc />
+    protected override void Initialize()
+    {
+        base.Initialize();
+
+        clipBoxSizeID = Shader.PropertyToID("_ClipBoxSize2");
+        clipBoxInverseTransformID = Shader.PropertyToID("_ClipBoxInverseTransform2");
+    }
+}
+
+// 2) Within MRTK/Core/StandardAssets/Shaders/MixedRealityStandard.shader (or another MRTK shader) add the following multi_compile pragma:
+
+#pragma multi_compile _ _CLIPPING_BOX2
+
+// 3) In the same shader change:
+
+#if defined(_CLIPPING_PLANE) || defined(_CLIPPING_SPHERE) || defined(_CLIPPING_BOX)
+
+// to:
+
+#if defined(_CLIPPING_PLANE) || defined(_CLIPPING_SPHERE) || defined(_CLIPPING_BOX) || defined(_CLIPPING_BOX2)
+
+// 4) In the same shader add the following shader variables:
+
+#if defined(_CLIPPING_BOX2)
+    fixed _ClipBoxSide2;
+    float4 _ClipBoxSize2;
+    float4x4 _ClipBoxInverseTransform2;
+#endif
+
```

Comment:
What are your thoughts of proposing this as add a second mechanism vs adding an array mechanism like here:
https://stackoverflow.com/questions/59930361/mrtk-multiple-clipping-boxes-for-one-object

If someone wants multiple clip primitives, it'll be easier to scale the array then do another re-write from clipBox, clipbox2, to clipbox3.

Reply:
I love your clipping box array solution, but I might stick to the current one to stay as succinct and grok-able as possible and leave the "elegant" solution as an exercise for the reader.